packages/polaris-middlewares/src/irrelevant-entities/irrelevant-entities-middleware.ts

```diff
@@ -27,25 +27,48 @@ export class IrrelevantEntitiesMiddleware {
         } as any;
     }
 
-    private static async queryIrrelevant(
-        connection: PolarisConnection,
-        tableName: string,
+    private async queryIrrelevant(
+        typeName: string,
         context: PolarisGraphQLContext,
         result: any,
     ) {
-        let irrelevantQuery = await connection
-            .getRepository(tableName)
-            .createQueryBuilderWithDeletedEntities(context, tableName)
-            .select('id');
+      if (this.connectionLessConfiguration) {
+        const irrelevantWhereCriteria: ConnectionlessIrrelevantEntitiesCriteria = {
+          realityId: context.requestHeaders.realityId || 0,
+          notInIds:
+              Array.isArray(result) && result.length > 0 ? result.map((x: any) => x.id) : [],
+          dataVersionThreshold: context.requestHeaders.dataVersion || 0,
+        }
+        return this.connectionLessConfiguration.getIrrelevantEntities(
+            typeName,
+            irrelevantWhereCriteria,
+        );
+      } else if(context?.requestHeaders?.realityId != null && this.connectionManager?.connections?.length) {
```

Comment:
this check happens before, so if there is a connectionless configuration it wont get inside this function